lib/Guacamole.pm

```diff
@@ -1377,12 +1377,58 @@ VersionNumberSegment ~ [0-9] [0-9] [0-9]
                      | [0-9]
 
 LitNumber ::= LitNumberDec
-LitNumberDec ~ Digits Period Digits
-             | Digits Period
-             | Period Digits
-             | Digits
+            | LitNumberOct
+            | LitNumberHex
+            | LitNumberBin
+
+LitNumberDec ::= NumberDec
+LitNumberOct ::= NumberOct
+LitNumberHex ::= NumberHex
+LitNumberBin ::= NumberBin
+
+NumberDec ~ NumberDecInt
+          | NumberDecInt ExpDec
+          | NumberDecInt '.' DigitsDec
+          | NumberDecInt '.' DigitsDec ExpDec
+          | '.' DigitDec DigitsDec
+          | '.' DigitDec DigitsDec ExpDec
+
+NumberDecInt ~ [1-9] DigitsDec
+             | '0'
```

Comment:
The treatment for `Underbars` is missing here, although maybe it is dealt with somewhere else (due to the need to exclude `0` for decimal digits)?

Reply:
It just seemed to me that the code for decimal numbers was different from the code for other types of numbers. For hex / oct / bin the rules obviously covered underscores, it wasn't obvious to me that the rules for decimals did too. But I also tried to say that maybe I was just me not seeing it.

I guess, bottom line, if the tests pass for stuff like `1_234_567` then everything is OK.

Reply:
Ah, good point. I have added more tests for these cases.